WordPress/src/main/java/org/wordpress/android/ui/comments/unified/UnifiedCommentListFragment.kt

```diff
@@ -167,7 +173,27 @@ class UnifiedCommentListFragment : Fragment(R.layout.unified_comment_list_fragme
     val commentDetails = registerForActivityResult(CommentDetailsActivityContract()) { response:
     CommentDetailsActivityResponse? ->
         if (response != null) {
-            viewModel.performSingleCommentModeration(response.commentId, response.commentStatus)
+            viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+                if (sharedPreferences?.getString(
```

Comment:
Since we are performing multiple actions here, personally I would bring all this logic into the VM, so we can test it with injected preferences provider.